maven-core/src/main/java/org/apache/maven/execution/MavenExecutionRequest.java

```diff
@@ -494,14 +504,49 @@ public interface MavenExecutionRequest {
 
     /**
      * @since 3.3.0
+     * @deprecated use {@link #setRootdir(Path)} instead
      */
+    @Deprecated
     void setMultiModuleProjectDirectory(File file);
 
     /**
      * @since 3.3.0
+     * @deprecated use {@link #getRootdir()} instead
      */
+    @Deprecated
     File getMultiModuleProjectDirectory();
 
+    /**
+     * Sets the root dir of the project.
+     *
+     * @since 4.0.0
+     */
+    MavenExecutionRequest setRootdir(Path rootdir);
+
+    /**
+     * Gets the root dir of the project, which is the up-most directory of the reactor, usually containing
+     * the {@code .mvn} directory.
+     *
+     * @since 4.0.0
+     */
+    Path getRootdir();
```

Comment:
This getter may fail (lazily) if invoked by there is no `.mvn` directory present (the value cannot be calculated)?

Reply:
If there's no `.mvn` directory, the value currently defaults to `topdir`, i.e. the current directory or the one pointed to with the `-f`/`--file` option. 
@cstamas we could change the behaviour to: print a warning, default to `null`, throw an exception when accessed using a getter (i.e. from plugins, or when injected / interpolated).  Is that what you have in mind ?